managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/cache/RangeEntryCacheImpl.java

```diff
@@ -276,14 +280,54 @@ public void asyncReadEntry(ReadHandle lh, long firstEntry, long lastEntry, boole
     private static final class ReadEntriesCallbackWithContext {
         final ReadEntriesCallback callback;
         final Object ctx;
+        final long startEntry;
+        final long endEntry;
     }
+
+    private CachedPendingRead findBestCandidate(PendingReadKey key, Map<PendingReadKey, CachedPendingRead> ledgerCache,
+                                                AtomicBoolean created) {
+        synchronized (ledgerCache) {
+            CachedPendingRead existing = ledgerCache.get(key);
+            if (existing != null) {
+                return existing;
+            }
+            // TODO: use a sorted map
```

Comment:
I am not sure we really need the SortedMap, it really depends on the size of the Map